Notes/BeetleBattleDesign.md

```diff
@@ -0,0 +1,115 @@
+
+### The Beetle
+
+Each beetle is a round sprite with a certain facing. They use "tank movement" where they can move forward and backwards
+based on their facing, and otherwise rotate their facing in either direction, clockwise or counter-clockwise. However,
+it is possible a trait gained through evolution could allow them to "strafe" or "dodge" directly to the left and right.
+
+Thinking of this from a physics modelling perspective, it would be ideal to keep track of a facing angle and a few basic
+velocity vectors:
+
+* Forward-backward velocity along the facing vector
+* Rotational velocity
+* Left-right sideways velocity perpendicular to the facing vector
+
+These desired vectors can be seen in this diagram:
+
+![The three desired vectors describing our beetle's physics: forward, sideways, and rotational velocities.](BeetleVectors.png)
+
+My first thought is that assuming the axes of `change_x` and `change_y` rotate along with `change_angle`, then both of
+those would be fine to use here as the x-axis and y-axis would thus be local to the beetle. However, from a code
+inspection it looks like those axes stay aligned with the global axes. While `change_angle` works OK for specifying
+rotational velocity, the other two velocities will require some calculation from the `angle` attribute.
+
+All the available sprite vector attributes are diagrammed here:
+
+![The four attributes describing the sprite: change_x, change_y, angle, and change_angle.](BeetleSpriteVectors.png)
+
+So the proposal is that we keep track of two scalar floats of speed, `forward_speed` and `sideways_speed`. Note that
+these are scalar, not vector values, but that's OK because so are the sprite attributes. Essentially, we will convert
+these speed values by finding the global x and y components of the vectors they describe based on `angle`. It might be
+useful to show this process on a diagram:
+
+![A diagram of our calculations of the global x and y components of our forward and sideways vectors.](BeetleSpriteVectors.png)
+
+While `forward_speed` and `sideways_speed` are scalars, the vectors they describe are drawn in green. We will have to
+calculate the purple vectors that are the x and y components. Luckily, we can use basic trigonometry here. Take notice
+of θ, which is 90 - `angle`. This makes sin(θ) equal to the length of the y-component of the forward vector divided
+the length of the length of the forwards vector. We can reason likewise for the x-component and cos(θ). Putting this
+together, and realizing that if θ is 90 - `angle`, then it is the other angle in the triangle, and thus we can just use
+`angle` directly:
+
+> `forward_x` = sin(`angle`) \* `forward_speed`
+> `forward_y` = cos(`angle`) \* `forward speed`
+> `sideways_x` = cos(`angle`) \* `sideways_speed`
+> `sideways_y` = sin(`angle`) \* `sideways_speed`
+> `change_x` = `forward_x` + `sideways_x`
+> `change_y` = `forward_y` + `sideways_y`
```

Comment:
If you actually look at this in the file, all of these get written into one paragraph because I forgot that Markdown does that by default if you do not end a line with two spaces.

I have a fix prepared.